app/jinja_filters.py

```diff
@@ -43,6 +43,29 @@ def format_date(value):
     return value.strftime('%-d %B %Y')
 
 
+@blueprint.app_template_filter()
+def format_conditional_date(dates):
```

Comment:
i think that if we make this method take 2 parameters then that would handle the None default for parameter 2.
you would just need to declare an additional `@blueprint.app_context_processor` for this